client/src/main/java/io/pravega/client/netty/impl/ClientConnectionImpl.java

```diff
@@ -49,20 +53,43 @@ public ClientConnectionImpl(String connectionName, int flowId, FlowHandler netty
     public void send(WireCommand cmd) throws ConnectionFailedException {
         checkClientConnectionClosed();
         nettyHandler.setRecentMessage();
-        Futures.getAndHandleExceptions(nettyHandler.getChannel().writeAndFlush(cmd), ConnectionFailedException::new);
+        write(cmd);
     }
 
     @Override
     public void send(Append append) throws ConnectionFailedException {
         Timer timer = new Timer();
         checkClientConnectionClosed();
         nettyHandler.setRecentMessage();
-        Futures.getAndHandleExceptions(nettyHandler.getChannel().writeAndFlush(append), ConnectionFailedException::new);
```

Comment:
The reason why we are waiting for the future to complete is that we need to be sure that the write and netty flush (not the tcp flush) completed. By waiting on the future the client avoid scenarios where `writeAndFLush` of the first append failed and the second append succeeded. 

Also, in the below write method 
```
if (eventLoop.inEventLoop()) {
            channel.write(cmd, promise); // if in event loop we enable batching of writes.
        }
```
we do not wait until promise completes. This would change the behavior of the client right? 🤔

Reply:
It will change the behaviour subtly, but we won't loose any exceptions. Note the line:
`promise.addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);`
This means that instead of the call failing during the call to write, the call will return successfully, but if the write actually fails, the failure handler will be invoked which will close the connection. 
This is effectively the same as if the data made it into the tcp buffer and then the connection dropped. We handle this by having acks and retry logic at upper layers. So it should be OK.